src/main/java/edu/gatech/cs2340/trydent/TrydentEngine.java

```diff
@@ -135,15 +154,50 @@ public static void quit() {
      * Hangs the current thread until the TrydentEngine stops running.
      */
     public static void waitUntilEngineStops() throws InterruptedException {
-        TrydentEngine engine = getInstance();
-        Thread waitOn = null;
-        synchronized (engine.mainLock) {
-            if (engine.mainThread == null || !engine.mainThread.isAlive())
-                return;
-            if (Thread.currentThread().equals(engine.mainThread))
-                return; // TODO: should probably throw an error.
-            waitOn = engine.mainThread;
+        // We don't have our own thread, because we rely on JavaFX
+        // to invoke the engine's update method, so we have to busy-wait.
+        while (getInstance().fxManager.isRunning()) {
+            try {
+                // Wait 100 ms and try again.
+                Thread.sleep(100);
+            } catch (InterruptedException ex) {
+
+            }
         }
-        waitOn.join();
     }
+
+    /**
+     * Returns true if the TrydentEngine has been started, and
+     * hasn't been stopped.
+     * @return
+     */
+    public static boolean isRunning() {
+        return getInstance().fxManager.isRunning();
+    }
+
+    /**
+     * Tells the TrydentEngine to run the given runnable
+     * on the next frame.
+     * @param runnable
+     */
+    public static void runLater(final Runnable runnable) {
+        new ContinuousEvent() {
```

Comment:
If it's getting stopped immediately, why is it being put into a continuous event?
If there's a reason, please add a comment.